sql/expression/convert.go

```diff
@@ -132,15 +132,21 @@ func (c *Convert) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 		return nil, nil
 	}
 
+	// Should always return nil, and a warning instead
 	casted, err := convertValue(val, c.castToType)
 	if err != nil {
-		return nil, ErrConvertExpression.Wrap(err, c.String(), c.castToType)
+		if c.castToType == ConvertToJSON {
+			return nil, ErrConvertExpression.Wrap(err, c.String(), c.castToType)
+		}
+		ctx.Warn(1292, "Incorrect %s value: %v", c.castToType, val)
```

Comment:
Could you add a quick test for this, too? I added some new fields to `ScriptTestAssertion` that make testing warnings really easy. Here's an example: https://github.com/dolthub/dolt/blob/main/go/libraries/doltcore/sqle/enginetest/dolt_queries.go#L1133